parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReader.java

```diff
@@ -0,0 +1,99 @@
+/* 
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.bytestreamsplit;
+
+import org.apache.parquet.bytes.ByteBufferInputStream;
+import org.apache.parquet.column.values.ValuesReader;
+import org.apache.parquet.io.ParquetDecodingException;
+
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class ByteStreamSplitValuesReader extends ValuesReader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ByteStreamSplitValuesReader.class);
+  private final int numStreams;
+  private final int elementSizeInBytes;
+  private byte[] byteStreamData;
+  private int indexInStream;
+  private int numberOfValues;
+
+  protected ByteStreamSplitValuesReader(int elementSizeInBytes) {
+    numStreams = elementSizeInBytes;
```

Comment:
Please add "this." to numStreams, indexInStream, numberOfValues to be consistent for all the fields.

Reply:
Done